recipes/root/all/conanfile.py

```diff
@@ -53,37 +51,100 @@ class RootConan(ConanFile):
         "libjpeg/9d",
         "libpng/1.6.37",
         "libcurl/7.73.0",
-        # "libuuid/1.0.3",
+        "pcre/8.44",
+        "xz_utils/5.2.5",
+        "zstd/1.4.5",
+        "lz4/1.9.3",
+        "glew/2.1.0",
+        "openssl/1.1.1h",
+        "fftw/3.3.8",
+        "cfitsio/3.490",
+        "tbb/2020.3",
+        "libpng/1.6.37",
     )
 
-    def configure(self):
-        tools.check_min_cppstd(self, "11")
+    def __init__(self, *args, **kwargs):
+        super(RootConan, self).__init__(*args, **kwargs)
+        self._cmake = None
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16.1",
+            "gcc": "4.8",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
 
     @property
-    def _rootsrcdir(self) -> str:
+    def _rootsrcdir(self):
         version = self.version.replace("v", "")
-        return f"root-{version}"
+        return "root-{}".format(version)
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn(
+                "{} recipe lacks information about the {} compiler support.".format(
+                    self.name, self.settings.compiler
+                )
+            )
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(
+                    "{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                        self.name,
+                        self._minimum_cpp_standard,
+                        self.settings.compiler,
+                        self.settings.compiler.version,
+                    )
+                )
 
     def source(self):
+        self._checkout_source()
+        self._fix_source_permissions()
+        self._patch_source_cmake()
```

Comment:
these need to be done in the build folder since they modify the sources (this is to enabled different conan workflows that consumes may use)

Reply:
> these need to be done in the build folder 

Do you mean this should be done in the build **method**? Assume that is correct this is done in c4a2736.